openmmtools/integrators.py

```diff
@@ -1895,6 +1895,232 @@ def _add_alchemical_reset_step(self):
+
+    @property
+    def _step_dispatch_table(self):
+        """
+        add an L step to the dispatch table
+        """
+        dispatch_table = super()._step_dispatch_table
+        dispatch_table['L'] = (self._add_backward_alchemical_perturbation_step, False)
+        return dispatch_table
+
+    def _add_backward_alchemical_perturbation_step(self):
+        """
+        _add_alchemical_perturbation_step except the lambda step is decremented
+        """
+        # Store initial potential energy
+        self.addComputeGlobal("Eold", "energy")
+
+        # Update lambda and increment that tracks updates.
+        self.addComputeGlobal('lambda', '(lambda_step-1)/n_lambda_steps')
+        self.addComputeGlobal('lambda_step', 'lambda_step - 1') #decrement instead of increment
+
+        # Update all slaved alchemical parameters
+        self._add_update_alchemical_parameters_step()
+
+        # Accumulate protocol work
+        self.addComputeGlobal("Enew", "energy")
+        self.addComputeGlobal("protocol_work", "protocol_work + (Enew-Eold)")
+
+    def get_forward_work(self, dimensionless=False):
+        """
+        pull the forward work from the integrator global variables
+        """
+        return self._get_energy_with_units('forward_work', dimensionless = dimensionless)
+
+    def get_backward_work(self, dimensionless=False):
+        """
+        pull the backward work from the integrator global variables
+        """
+        return self._get_energy_with_units('backward_work', dimensionless=dimensionless)
+
+class LoopyIndependentAlchemicalNonequilibriumLangevinIntegrator(LoopyAlchemicalNonequilibriumLangevinIntegrator):
+    """
+    Subclass of LoopyAlchemicalNonequilibriumLangevinIntegrator that runs two independent equilibrium simulations at alchemical
+    for annealing purposes.
+    """
+    def __init__(self,
+                 zero_endstate_positions,
+                 one_endstate_positions,
+                 alchemical_functions = None,
+                 nsteps_eq = 1000,
+                 nsteps_neq = 100,
+                 **kwargs):
+        """
+        arguments
+            zero_endstate_positions : np.ndarray() (N, 3) unit.Quantity (in length units)
+                starting positions of the zero endstate equilibration
+            one_endstate_positions : np.ndarray() (N,3) unit.Quantity (in length units)
+                starting positions of the one endstate equilibration
+        """
+        super().__init__(alchemical_functions = alchemical_functions,
+                         nsteps_eq = nsteps_eq,
+                         nsteps_neq = nsteps_neq,
+                         **kwargs)
+
+        self.addPerDofVariable('x_zero_endstate', zero_endstate_positions.value_in_unit_system(unit.md_unit_system))
+        self.addPerDofVariable('x_one_endstate', zero_endstate_positions.value_in_unit_system(unit.md_unit_system))
+        self.addPerDofVariable('x_annealing', 0.)
```

Comment:
Is `x_annealing` needed?

I think you could replace
```python
self.addComputePerDof('x_annealing', 'x_zero_endstate')
self.addComputePerDof('x', 'x_annealing')
```
etc. with `self.addComputePerDof('x', 'x_zero_endstate')` directly, and I think `self.addComputePerDof('x_annealing', 'x_one_endstate')` may not be referenced later

Reply:
@dominicrufa : Can you address this?

Reply:
`x_annealing` is redundant; @maxentile  is right, i just forgot to delete.  but this modification is for the second integration scheme where we have two independent MD runs at the endstates, which is not currently implemented or tested

Reply:
fixed now